c++/src/kj/async-io-test.c++

```diff
@@ -176,8 +176,10 @@ TEST(AsyncIo, SimpleNetworkAuthentication) {
 TEST(AsyncIo, UnixSocket) {
   auto ioContext = setupAsyncIo();
   auto& network = ioContext.provider->getNetwork();
-
-  auto path = kj::str(TMPDIR "/kj-async-io-test.", getpid());
+ 
+  const char* tmpDir_env = getenv("TEST_TMPDIR");
+  const auto tmpDir = tmpDir_env != nullptr ? tmpDir_env : TMPDIR;
+  auto path = kj::str(tmpDir, "/kj-async-io-test.", getpid());
```

Comment:
Unfortunately this change will likely break bazel users as bazel's TEST_TMPDIR is often a very long path, and unix sockets have a length limit of 104 characters.

This change isn't described in the PR description, was this intended to be part of the PR?